src/mango_httpd.erl

```diff
@@ -50,8 +50,24 @@ handle_req_int(_, _) ->
 
 
 handle_index_req(#httpd{method='GET', path_parts=[_, _]}=Req, Db) ->
+    Params = lists:flatmap(fun({K, V}) -> parse_index_param(K, V) end,
+        chttpd:qs(Req)),
     Idxs = lists:sort(mango_idx:list(Db)),
-    JsonIdxs = lists:map(fun mango_idx:to_json/1, Idxs),
+    JsonIdxs0 = lists:map(fun mango_idx:to_json/1, Idxs),
+    % Set limit default to 200 here since dashboard defaults to 200 per page.
+    Limit = case couch_util:get_value(limit, Params, 200) of
+        Limit0 when Limit0 < 1 ->
+            ?MANGO_ERROR(invalid_list_index_params);
+        Limit0 ->
+            Limit0
+    end,
+    Skip = case couch_util:get_value(skip, Params, 0) of
+        Skip0 when Skip0 < 0; Skip0 > length(JsonIdxs0) ->
```

Comment:
I think here is a broken edge case. Assume the following case:
- There are 16 indexes in DB and they are paged by factor 5
- UserA browses page 3 (skip=10, limit=5)
- UserB deletes two indexes
- UserA goes on the next page (skip=15, limit=5)

Is it fair to reply with HTTP 500 to UserA instead of empty list when client have to skip more items than available?